[yimjianbing] iP

yimjianbing opened the pull request:
### Wen Jie <3

> "ride my banana boat sometime??" – Wen Jie ([source](https://vt.tiktok.com/ZS2yuednG))

Ever wanted a personal femboy maid that could talk to you during those oh so lonely nights? 
Well look no further than the newly founded WEN JIE BOT! 

Here's a list of actions that the Wen Jie bot can perform!

- [ ]  manage your deadlines
- [ ]  manage your todos
- [ ]  manage your events
- [ ]  mark tasks as completed or not

Interested? I know I am! Here's how to set up Wen Jie on your system! 🥰🥰

1. download it from [here](https://github.com/yimjianbing/ip/releases/tag/A-Jar)
2. double-click it
3. be mesmerised by the amazing bot and maid outfit
4. sit back as Wen Jie ~seduce~ keep track of tasks for you!

Still not convinced? Here's some benefits if you were to install Wen Jie on your laptop right now!

- You get max aura
- Your goon level increases by twenty
- skibidi ohio rizz

WHAT ARE YOU WAITING FOR?! GET YOUR PERSONAL MAID CHATBOT RIGHT NOW!!!!!!!

OH and you can learn java if you want to i guess, if thats your thing, the **main** method is here:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yimjianbing committed:
Renamed Duke to Wen Jie and added his greeting + exit

yimjianbing committed:
Fixed the class name to be Wen Jie, and added a missing semi colon

yimjianbing committed:
Personalised chatbot further

yimjianbing committed:
Added echo for ChatBot

yimjianbing committed:
Added add function and list to chatbot

yimjianbing committed:
Added a Task Class and overridden the toString method

yimjianbing committed:
Modified code to use the Task Object when calling list instead of with String

yimjianbing committed:
Finished up the switch cases for mark and unmark and fixed the logic

yimjianbing committed:
Added Todos, Events and Deadline classes

yimjianbing committed:
Added logic for Event class

yimjianbing committed:
Finished logic for deadline, event and todos

yimjianbing committed:
Set up semi-autmoated testing for the Ui

yimjianbing committed:
Added exceptions and their respective logic for edge cases

yimjianbing committed:
Changed output for OutOfBoundsException

yimjianbing committed:
Changed data type of tasklist from an array of tasks to an arraylist of tasks, and adde dnew exception for invalid input for event/deadline

yimjianbing committed:
Added logic for delete command, along with a new exception called NoNumberInputtedException

yimjianbing committed:
Add wenjie.txt and Storage.java

yimjianbing committed:
Finish implementation of Storage class and logic handling

yimjianbing committed:
Integrate Storage class into Wenjie.java

yimjianbing committed:
Fix bugs and made storage actually store data, as well as read from it

yimjianbing committed:
Merge branch 'branch-Level-7'

yimjianbing committed:
Change Deadline.java to use LocalDateTime to store deadline with changed
formatting

yimjianbing committed:
Merge branch 'branch-Level-8'

yimjianbing committed:
Add both Ui.java and TaskList.java as instructed

yimjianbing committed:
Add commands for Ui and refactored WenJie.java

yimjianbing committed:
Add Command abstract class, and its children class

yimjianbing committed:
Add and moved the different commands, and begun implementation of the
isExit boolean in all command children classes

yimjianbing committed:
Add parser and added logic for most command classes

yimjianbing committed:
Refactor switch statements from WenJie.java into its own class

yimjianbing committed:
Edit output messages to make it neater and refined logic

yimjianbing committed:
Merge branch 'A-MoreOOP'

yimjianbing committed:
Rearrange files into organized packages called exceptions and tasks

yimjianbing committed:
Merge branch 'A-Packages'

yimjianbing committed:
Merge branch 'add-gradle-support'

yimjianbing committed:
Change toString() in the exceptions to override getMessage instead, and
properly integrated gradle to be ablet to run WenJie.java

yimjianbing committed:
Fix bug regarding deadline not parsing properly

yimjianbing committed:
Fix major bug where Storage could not read deadline from wenjie.txt

yimjianbing committed:
Finish up test cases, and fixed the typo for isActive to isExit in
ByeCommand

yimjianbing committed:
Add JUnit tests for the different tasks

yimjianbing committed:
Merge branch 'A-JUnit'

yimjianbing committed:
Package app as a .jar file

yimjianbing committed:
Add FindCommand and its logic, along with its own switch case in
Parser

yimjianbing committed:
Add javadocs for Command.java

yimjianbing committed:
Add javadocs for Command.java

yimjianbing committed:
Change all switch statements to have no indentations

yimjianbing committed:
Fix import statements to no longer use wildcard imports

yimjianbing committed:
Add javadocs for all java classes except test classes

yimjianbing committed:
Add javadocs for FindCommand.java

yimjianbing committed:
Merge branch 'A-Level9'

yimjianbing committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/wenjieBot/Parser.java

yimjianbing committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	src/main/java/wenjieBot/Parser.java
#	src/main/java/wenjieBot/Storage.java
#	src/main/java/wenjieBot/commands/AddCommand.java
#	src/main/java/wenjieBot/tasks/Task.java

skyl3-r commented on src/main/java/wenjieBot/Parser.java:
```diff
@@ -0,0 +1,65 @@
+package wenjieBot;
+
+import wenjieBot.commands.*;
```
Imported classes should be listed explicitly instead of using *, in order to make the class easier to comprehend and maintain.

skyl3-r commented on src/main/java/wenjieBot/Storage.java:
```diff
@@ -0,0 +1,165 @@
+     * @param isDone whether the task is marked as done.
+     * @return the corresponding Task object.
+     */
+    private static Task getTask(String stringifiedTask, boolean isDone) {
```
Maybe this method could be called convertStringToTask instead of getTask to make what it does clearer? Argument names look good though!

skyl3-r commented on src/main/java/wenjieBot/Ui.java:
```diff
@@ -0,0 +1,66 @@
+     *
+     * @param scanner the Scanner object used to read input from the user.
+     * @return the command entered by the user as a string.
+     */
```
Overall your javadocs look quite well written, good job!

yastsc replied:
I agree, your JavaDocs are really thorough and informative!

skyl3-r commented on src/main/java/wenjieBot/exceptions/DukeException.java:
```diff
@@ -0,0 +1,14 @@
+package wenjieBot.exceptions;
+
+public class DukeException extends Exception{
```
I think DukeException can be renamed to WenjieException to make it clearer 👍

skyl3-r commented on src/main/java/wenjieBot/exceptions/InvalidInputException.java:
```diff
@@ -0,0 +1,14 @@
+package wenjieBot.exceptions;
+
+public class InvalidInputException extends DukeException {
```
Seems like this exception is only for date inputs, perhaps it can be renamed to InvalidDateInputException instead?

skyl3-r reviewed (commented):
Overall the code looks good, just some minor comments and things to change. Seems like the testing is quite comprehensive, so good job!

yastsc commented on src/main/java/wenjieBot/WenJie.java:
```diff
@@ -0,0 +1,63 @@
+package wenjieBot;
```
Since names representing packages should be in all lowercase, perhaps naming it wenjiebot instead of wenjieBot would be better.

yastsc commented on src/main/java/wenjieBot/Parser.java:
```diff
@@ -0,0 +1,65 @@
+        String[] parts = input.split(" ");
+        String action = parts[0].toLowerCase();
+
+        switch (action){
```
There should be a white space between (action) and {, as Java reserved words should be followed by a white space.

yastsc commented on src/main/java/wenjieBot/TaskList.java:
```diff
@@ -0,0 +1,38 @@
+    /**
+     * Constructs an empty TaskList.
+     */
+    public TaskList() {
+        // No tasks initialized
+    }
```
Might be better to initialise the new empty ArrayList here in the constructor for the empty TaskList instead of above at the declaration.

yastsc commented on src/main/java/wenjieBot/commands/AddCommand.java:
```diff
@@ -0,0 +1,192 @@
+package wenjieBot.commands;
+
+import wenjieBot.Storage;
+import wenjieBot.Ui;
+import wenjieBot.TaskList;
```
The TaskList import statement should be above the Ui import statement as the ordering of import statements needs to be consistent, in alphabetical order.

yastsc commented on src/main/java/wenjieBot/commands/AddCommand.java:
```diff
@@ -0,0 +1,192 @@
+ * Represents a command to add a task to the task list.
+ * The task can be a ToDo, Deadline, or Event based on the type of event specified.
+ */
+public class AddCommand extends Command {
```
Good job following PascalCase for class names and CamelCase for variable names!

yastsc reviewed (commented):
Overall, your JavaDocs are quite impressive and you've consciously followed most of the coding standard. There are just a few minor changes you can make to align your code to the coding standard even more :)

yimjianbing committed:
Refactor code by renaming package and adding checkstyle plugin

yimjianbing committed:
debug tests and finish code restyling for entire codebase

yimjianbing committed:
Merge branch 'A-CheckStyle'

yimjianbing committed:
Redo commands to store and return a string to display in GUI instead
of using println directly

yimjianbing committed:
Add css and finish javafx tutorial

yimjianbing committed:
Add padding for dialog box

yimjianbing committed:
Add javadocs for exceptions and gui classes

yimjianbing committed:
Fix coding style

yimjianbing committed:
fix and change style

yimjianbing committed:
Merge branch 'branch-Level-10'

yimjianbing committed:
Add varargs

yimjianbing committed:
Merge branch 'A-Varargs'

yimjianbing committed:
Change avatar and personalised the messages

yimjianbing committed:
Change method and class names according to PR comments

yimjianbing committed:
Changed colors of send button in css

yimjianbing committed:
Added better logic check for parsing

yimjianbing committed:
Fix parsing for deadline and event, as well as change event to use java
data time library as well

yimjianbing committed:
Add an exit method for when the user calls the bye command

yimjianbing committed:
Edit build.gradle to allow usage of shadowjar plugin

To integrate the usage of the shadowjar plugin to create a fat JAR file,
the build.gradle's dependencies had to be modified.

Specifically added the lines mainClassName = 'wenjiebot.WenJie' so that
the plugin knows where the entry point of the programme is, to ensure
the plugin runs properly.

It was done this way as given in the instructions on the se-edu website.

yimjianbing committed:
Refactor code in Storage.java

To keep in line with the coding standard of SLAP, and to decrease the
number of lines of code within the methods to less than 30.

By using IntelliJ's refactoring feature, I abstracted out some parts of
each method to its own new method to follow the code quality.

Let's try to ensure that we use more guard clauses.

yimjianbing committed:
Refactor Add Command

yimjianbing committed:
Fix bug regarding deadlines and refactored

yimjianbing committed:
Add assert statements in mark and unmark commands

yimjianbing committed:
Change WenjJie to use stream for the chatbot loop

yimjianbing committed:
Merge pull request #2 from yimjianbing/branch-A-Streams

Change WenjJie to use stream for the chatbot loop

yimjianbing committed:
Merge branch 'master' into branch-A-Assertions

yimjianbing committed:
Merge pull request #3 from yimjianbing/branch-A-Assertions

Branch A Assertions

yimjianbing committed:
Merge branch 'master' into branch-A-CodeQuality

yimjianbing committed:
Merge pull request #4 from yimjianbing/branch-A-CodeQuality

Branch a code quality

yimjianbing committed:
Add more assert and fix ParserTest

yimjianbing committed:
Add gradle.yml

yimjianbing committed:
Fix errors with outdated JUnit Tests

yimjianbing committed:
Remove UiTest and fix stlying

yimjianbing committed:
Edit runtest.sh

yimjianbing committed:
Edit runtest.sh to fit new code

yimjianbing committed:
Delete unneeded segments of gradle.yml used for io tests

yimjianbing committed:
Merge branch 'branch-A-CI'

yimjianbing committed:
Add snooze feature

yimjianbing committed:
Merge branch 'branch-B-Snooze'

yimjianbing committed:
Add Invalid Snooze Form exception

yimjianbing committed:
Add SnoozeOnTodoException

yimjianbing committed:
Add javadocs for new methods and classes

yimjianbing committed:
Merge branch 'A-MoreErrorHandling'

yimjianbing committed:
Edit some ui messages to give the bot more personality

yimjianbing committed:
Merge branch 'branch-A-Personality'

yimjianbing committed:
Add background to chatbot

yimjianbing committed:
Merge branch 'branch-A-BetterGUI'

yimjianbing committed:
Fix checkstyle issues

yimjianbing committed:
Change javadocs to fit standard

yimjianbing committed:
Update ReadME.md in /docs

yimjianbing committed:
Merge branch 'branch-A-UserGuide'

yimjianbing committed:
Finalise Readme.md

yimjianbing committed:
Edit UI.png name to Ui.png

yimjianbing committed:
Fix errors regrading welcome message not shown and add more exception

yimjianbing committed:
Fix style errors and add more exceptions handling
